### ISSUE TITLE
Do not deploy HTML files if a build fails

### ISSUE THREAD

**Issue by shinaoka** [tensor4all/juliaexamples#11]
We do not want to deploy broken html files.

**Comment by terasakisatoshi** [comment]
Adding [`--warningiserror`](https://jupyterbook.org/en/stable/reference/cli.html#cmdoption-jupyter-book-build-W) option in scripts/jupyter_book.jl should solve our issue.

<img width="996" alt="image" src="https://github.com/tensor4all/T4AJuliaTutorials/assets/16760547/48035b0e-0c6f-49da-88d8-e759023bff2d">

This will treat warnings shown in GH Actions as errors.

<img width="1597" alt="image" src="https://github.com/tensor4all/T4AJuliaTutorials/assets/16760547/5af07f91-529f-45ea-aa91-2b9fb14f3d86">

In short, by running `jupyter-book build . --warningiserror` we will get:

<img width="1045" alt="image" src="https://github.com/tensor4all/T4AJuliaTutorials/assets/16760547/f3987c65-1fe0-45a2-bcbd-48c42f6020cc">
